web/modules/custom/nlc_prototype/src/Plugin/Block/MyCohortBlock.php

```diff
@@ -42,8 +42,8 @@ public function build() {
           $entity = \Drupal::entityTypeManager()
             ->getStorage('node')
             ->load($cohort->target_id);
-          $options['query']['directory'][] = 'cohort:' . $cohort->target_id;
-          if (!empty($options)) {
+          $options['query']['f'][] = 'cohort:' . $entity->label();
+          if ($entity && !empty($options)) {
```

Comment:
Basically, this all looks great, but just one tiny question here. The `if` test includes the entity object, but in the line above the `$entity->label()` method has just been used — so this tests for the existence of something that's already been  assumed to exist. Which is a bit odd.

Worth tidying up?